jazz_environments/components/validation/validate_create_payload.js

```diff
@@ -0,0 +1,190 @@
+            //check for empty body
+            logger.info("Inside validateIsEmptyInputData: ");
+            validateUtils.validateIsEmptyInputData(environment_data, onComplete)
+        },
+
+        validateAllRequiredFields: function(onComplete){
+            //check for required fields
+            logger.info("Inside validateAllRequiredFields: ");
+            validateUtils.validateAllRequiredFields(environment_data, required_fields_create, onComplete);
+        },
+
+        validateStatusFieldValue: function(onComplete) {
+            //check for valid status 
+            logger.info("Inside validateStatusFieldValue: ");
+            validateUtils.validateAllRequiredFieldsValue(environment_data, global.config.service_environment_required_fields, onComplete);
+        },
+
+        validateUnAllowedFieldsInInput: function(onComplete) {
+            //check for unchangable fields
+            logger.info("Inside validateUnAllowedFieldsInInput: ");
+            validateUtils.validateUnAllowedFieldsInInput(environment_data, service_field_list, onComplete);
+        },
+
+        validateFriendlyName: function(onComplete){
+            //check for friendly name
+            logger.info("Inside validateFriendlyName: "+JSON.stringify(environment_data));
+            validateUtils.validateFriendlyName(environment_data, onComplete);
+        },
+
+        validateAllRequiredFieldsValue: function(onComplete) {
+            //check for required fields
+            logger.info("Inside validateAllRequiredFieldsValue: ");
+            validateUtils.validateAllRequiredFieldsValue(environment_data, required_fields_create, onComplete);
+        },
+
+        validateRemoveEmptyValues: function(onComplete) {
+            // check for empty values before updating environments table
+            logger.info("Inside validateRemoveEmptyValues: ");
+            validateUtils.validateRemoveEmptyValues(environment_data, onComplete);
+        },
+        // Check if a service with same domain and service_name combination exists
+        validateServiceExists: function(onComplete) {
+            logger.info("validateServiceExists: ");
+            var service_domain = environment_data.domain;
+            var service_name = environment_data.service;
+            var svcGetPayload;
+            //call services get
+            svcGetPayload = {
+                uri: global.config.SERVICE_API_URL + global.config.SERVICE_API_RESOURCE + "?domain=" + service_domain + "&service=" + service_name,
+                method: "GET",
+                headers: {'Authorization':global.authorization},
+                rejectUnauthorized: false
+            };
+            request(svcGetPayload, function(error, response, body) {
+                if (response.statusCode === 200) {
+                    var output = JSON.parse(body);
+                    if (output.data === null || output.data === "" || output.data === undefined || output.data.available === undefined) {
```

Comment:
use short form `if (!output || !output.data || !output.data. available) { `

Reply:
Worked on the changes